src/Neo/SmartContract/ApplicationEngine.Runtime.cs

```diff
@@ -394,9 +399,16 @@ protected internal void RuntimeNotifyV1(byte[] eventName, Array state)
         /// <param name="state">The arguments of the event.</param>
         protected internal void SendNotification(UInt160 hash, string eventName, Array state)
         {
+            notifications ??= new List<NotifyEventArgs>();
+            // Restrict the number of notifications for Application executions. Do not check
+            // persisting triggers to avoid native persist failure. Do not check verification
+            // trigger since verification context is loaded with ReadOnly flag.
+            if (IsHardforkEnabled(Hardfork.HF_Echidna) && Trigger == TriggerType.Application && notifications.Count >= MaxNotificationCount)
```

Comment:
@AnnaShaleva We should add Verification
```suggestion
            if (IsHardforkEnabled(Hardfork.HF_Echidna) && (Trigger == TriggerType.Application || Trigger == TriggerType.Verification) && notifications.Count >= MaxNotificationCount)
```

Reply:
Doubt that. We use `CallFlags.ReadOnly` for verifications.